crates/notion-core/src/shell/mod.rs

```diff
@@ -73,24 +60,16 @@ impl Shell for CurrentShell {
     }
 }
 
-/// Thrown when the shell name specified in the Notion environment is not supported.
-#[derive(Debug, Fail, NotionFail)]
-#[fail(display = "Unrecognized command shell name: {}", name)]
-#[notion_fail(code = "EnvironmentError")]
-struct UnrecognizedShellError {
-    name: String,
-}
-
 impl FromStr for CurrentShell {
     type Err = NotionError;
 
     fn from_str(src: &str) -> Result<Self, NotionError> {
-        let postscript_path = env::postscript_path().ok_or(UnspecifiedPostscriptError)?;
+        let postscript_path = env::postscript_path().ok_or(ErrorDetails::UnspecifiedPostscript)?;
```

Comment:
This needs a merge from `master`, it looks like: the logic here doesn't include the changes I made for fish support.